File: Cats_And_Mouses/cats_and_mouses.rb
def catAndMouse(x, y, z)
  units_from_mouse_a = z > x ? (z - x) : ((z - x) * -1)
  units_from_mouse_b = z > y ? (z - y) : ((z - y) * -1)

  if units_from_mouse_a == units_from_mouse_b
    'Mouse C'
  elsif units_from_mouse_a > units_from_mouse_b
    'Cat B'
  else
    'Cat A'
  end
end

File: Cats_And_Mouses/catsAndMouses.js
// Complete the catAndMouse function below.
function catAndMouse(x, y, z) {
    const CAT_A_DISTANCE = (z > x) ? (z - x) : ((z - x) * -1);
    const CAT_B_DISTANCE = (z > y) ? (z - y) : ((z - y) * -1);

    if (CAT_A_DISTANCE === CAT_B_DISTANCE){
        return 'Mouse C';
    } else if (CAT_A_DISTANCE > CAT_B_DISTANCE) {
        return 'Cat B';
    } else {
        return 'Cat A';
    }
}